CMakeLists.txt

```diff
@@ -309,10 +309,15 @@ find_package(Perl)
 option(BUILD_DOCS "to build manual pages" ON)
 option(ENABLE_MANUAL "to provide the built-in manual" OFF)
 
-if(ENABLE_MANUAL AND PERL_FOUND)
-  curl_nroff_check()
-  if(NROFF_USEFUL)
-    set(USE_MANUAL ON)
+if(ENABLE_MANUAL)
+  if(BUILD_DOCS AND PERL_FOUND)
+    curl_nroff_check()
+    if(NROFF_USEFUL)
+      set(USE_MANUAL ON)
+    endif()
+  endif()
+  if(NOT USE_MANUAL)
+    message(WARNING "BUILD_DOCS disabled, or perl not found, or nroff not useful.  Will not provide built-in manual")
```

Comment:
This seems to also disable building curl.1 and tool_hugehelp.c (aka the 'manual') when
`BUILD_DOCS=OFF`. The point of adding `BUILD_DOCS` was to have a separate
control for the rest of docs (the individual man pages.), which this patch appears to 
break.

Reply:
That's quite confusing, especially since `BUILD_DOCS` is documented as "to build manual pages", with no other qualification...  so what makes curl.1 a manual page that gets built even though building docs has been disabled.  This makes no sense at all.

We can see it another way, though: if it were me, I would remove `BUILD_DOCS`, i.e. always build the docs.  Would that be a suitable option?

Reply:
The wording is confusing indeed, but note that up until a few days ago none of these options worked correctly with CMake.

We have a manual page (and a built-in copy) for the curl tool and manual pages for libcurl.

I added `BUILD_DOCS` because building them took a lot of time. I've made it faster, but it still takes quite a bit of extra time. Also some builds (e.g. curl-for-win) doesn't distribute or need the man pages, so building them is just wasting CI time. I'd like to keep this option. It's also enabled by default.

Here's my proposal to separate these two settings, and to make the wording clearer: #12773